### PR TITLE
Remove Appveyor in favor of Travis Windows

### PR DESCRIPTION
**Pull request by Osspial**
Now that Travis supports Windows, migrating everything to the same testing infrastructure will reduce complexity.

Travis is currently advertising its Windows builds as "early stage", so we might want to hold off on merging this until they're advertising Windows builds as stable.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,38 @@ matrix:
...
+    # Windows GNU 32bit
+    - env: TARGET=i686-pc-windows-gnu
+      os: windows
+      rust: nightly
+    - env: TARGET=i686-pc-windows-gnu
+      os: windows
+      rust: stable
+
+    # Windows MSVC 64bit
+    - env: TARGET=x64_64-pc-windows-msvc
```
Comment (ZeGentzy) [inline .travis.yml]: 86 goes before the 64.

Comment (felixrabe) [reply .travis.yml]: I.e. should read: `x86_64`

Comment (Osspial) [reply .travis.yml]: Whoops. Fixed.

File: appveyor.yml [hunk 2]
```diff
@@ -1,23 +0,0 @@
-environment:
```
Comment (ZeGentzy) [inline appveyor.yml]: For some reason appveyor is still trying to build- then failing. Might want to figure out how to disable that.

File: .travis.yml [hunk 3]
```diff
@@ -26,6 +26,38 @@ matrix:
...
+    # Windows MSVC 32bit
+    - env: TARGET=i686-pc-windows-msvc
+      os: windows
+      rust: nightly
+    - env: TARGET=i686-pc-windows-msvc
+      os: windows
+      rust: stable
+
+    # Windows GNU 32bit
+    - env: TARGET=i686-pc-windows-gnu
```
Comment (ZeGentzy) [inline .travis.yml]: This mingw build is failing.

File: .travis.yml [hunk 4]
```diff
@@ -26,6 +26,38 @@ matrix:
...
+    # Windows MSVC 64bit
+    - env: TARGET=x86_64-pc-windows-msvc
+      os: windows
+      rust: nightly
+    - env: TARGET=x86_64-pc-windows-msvc
+      os: windows
+      rust: stable
+
+    # Windows GNU 64bit
+    - env: TARGET=x86_64-pc-windows-gnu
```
Comment (ZeGentzy) [inline .travis.yml]: This mingw build is failing.

File: appveyor.yml (hunk 2 above)
Comment (Osspial) [reply appveyor.yml]: Appveyor is installed into this github repository, so we'll just have to remove that from the repo's control panel.

**Comment by rukai** [comment]
Definitely do not merge this until they fix secrets on windows.
Windows builds currently fail when secrets are enabled.
It will build PR's fine but will fail when building the master branch.
I made this mistake on vulkano
https://travis-ci.community/t/timeout-before-install-step/1502

**Comment by Osspial** [comment]
Closing in favor of #1309

**Pull request closed without merging**